File: cognipy/CogniPyUnitTests/CnlManipulationTests.cs
using CogniPy;
using NUnit.Framework;
using Ontorion.FluentEditorClient;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CogniPyUnitTests
{
    [TestFixture]
    public class CnlManipulationTests
    {
        [Test]
        public void AddCnl()
        {
            var feClient = new CogniPySvr();
            // reason the new context
            var CnlContent = "Every man is a human-being.\r\n Every human-being has-name equal-to 'aaa'.";
            var CnlToAdd = new List<string>() {
                    "John is a man.",
                    "Every man is a human-being."
                };

            feClient.LoadCnlFromString(CnlContent, true, true);
            feClient.KnowledgeInsert(string.Join("\r\n", CnlToAdd), true, true);
            var mergedCnl = feClient.ToCNLList(false);

            foreach (var cnl in CnlToAdd)
            {
                Assert.IsTrue(mergedCnl.Any(x => x == cnl));
            }
            Assert.AreEqual(3,mergedCnl.Count());
        }

        [Test]
        public void AddSubsumptionTop()
        {
            var feClient = new CogniPySvr();

            try
            {
                feClient.LoadCnlFromString("My-Instance-1 is a thing.", true, true);
                feClient.KnowledgeInsert("My-Instance-2 is a thing.", true, true);
                feClient.KnowledgeInsert("Every my-concept is a thing.",true,true);
            }
            catch(Exception ex)
            {
                Assert.Fail(ex.Message);
            }
        }

        [Test]
        public void AddNumberCnl()
        {
            var feClient = new CogniPySvr();
            // reason the new context
            var CnlContent = "Vendor-0 is a vendor.";
            var CnlToAdd = new List<string>() {
                    "Vendor-0 has-latitude equal-to 43.737345.",
                    "Vendor-0 has-longitude equal-to -79.442286."
                };

            feClient.LoadCnlFromString(CnlContent, true, false);
            feClient.KnowledgeInsert(string.Join("\r\n", CnlToAdd), true, false);

            var mergedCnl = feClient.ToCNLList(false);

            foreach (var cnl in CnlToAdd)
            {
                Assert.IsTrue(mergedCnl.Any(x => x == cnl));
            }
            Assert.AreEqual(3, mergedCnl.Count());
        }
        [Test]
        public void RemoveCnl()
        {
            var feClient = new CogniPySvr();
            // reason the new context
            var CnlContent = "John is a man.\r\nEvery man is a human-being.\r\n Every human-being has-name equal-to 'aaa'.\r\n";
            var CnlToRemove = new List<string>() {
                    "John is a man."
                    //,                    "Every man is a human-being."
                };

            feClient.LoadCnlFromString(CnlContent, true, true);
            feClient.KnowledgeDelete(string.Join("\r\n", CnlToRemove), false);
            var mergedCnl = feClient.ToCNLList(true,true,true);

            foreach (var cnl in CnlToRemove)
            {
                Assert.IsTrue(!mergedCnl.Any(x => x == cnl));
            }
            Assert.AreEqual(2, mergedCnl.Count());
        }

        [Test]
        public void RulesKeptAfterMergeTest()
        {
            var feClient = new CogniPySvr();
            // reason the new context
            var CnlContent = new List<string>() {
                "If a man is a human-being then the man is a cat."
            };
            var CnlToAdd = new List<string>() {
                    "John is a man."
                };

            feClient.LoadCnlFromString(string.Join("\r\n",CnlContent), true, true);
            feClient.KnowledgeInsert(string.Join("\r\n", CnlToAdd), true, true);
            var mergedCnl = feClient.ToCNLList(true, true, true);

            Assert.AreEqual(2, mergedCnl.Count());
            foreach(var cnl in CnlContent)
            {
                Assert.IsTrue(mergedCnl.Any(x => x == cnl));
            }

            foreach (var cnl in CnlToAdd)
            {
                Assert.IsTrue(mergedCnl.Any(x => x == cnl));
            }

        }

        [Test]
        [TestCase("John", "instance")]
        [TestCase("man", "concept")]
        [TestCase("has-name", "role")]
        [TestCase("John[TIS]", "instance")]
        public void FromToUriTest(string cnlName,string type)
        {
            var feClient = new CogniPySvr();
            feClient.LoadCnlFromString("John is a man.\r\nReferences: [TIS] (\"http://www.somenamespace.com/\").",true,false);
            
            var uri = feClient.UriFromCnl(cnlName, type);
            var transformedCnl = feClient.CnlFromUri(uri, type);

            Assert.AreEqual(cnlName, transformedCnl);
        }
    }
}

File: cognipy/cognipytest/cognipyunittests/CnlManipulationTests.java
package cognipyunittests;

import CogniPy.*;
import NUnit.Framework.*;
import java.util.*;

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//ORIGINAL LINE: [TestFixture] public class CnlManipulationTests
public class CnlManipulationTests
{
//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//ORIGINAL LINE: [Test] public void AddCnl()
	public final void AddCnl()
	{
		CogniPySvr feClient = new CogniPySvr();
		// reason the new context
		String CnlContent = "Every man is a human-being.\r\n Every human-being has-name equal-to 'aaa'.";
		ArrayList<String> CnlToAdd = new ArrayList<String>(Arrays.asList("John is a man."));

		feClient.LoadCnlFromString(CnlContent, true, true, false);
		feClient.KnowledgeInsert(tangible.StringHelper.join("\r\n", CnlToAdd), true, true);
		ArrayList<String> mergedCnl = feClient.ToCNLList(false);

		for (String cnl : CnlToAdd)
		{
			assert mergedCnl.Any(x = cnl.equals(> x));
		}
		assert 3 == mergedCnl.size();
	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//ORIGINAL LINE: [Test] public void AddNumberCnl()
	public final void AddNumberCnl()
	{
		CogniPySvr feClient = new CogniPySvr();
		// reason the new context
		String CnlContent = "Vendor-0 is a vendor.";
		ArrayList<String> CnlToAdd = new ArrayList<String>(Arrays.asList("Vendor-0 has-latitude equal-to 43.737345.", "Vendor-0 has-longitude equal-to -79.442286."));

		feClient.LoadCnlFromString(CnlContent, true, false, false);
		feClient.KnowledgeInsert(tangible.StringHelper.join("\r\n", CnlToAdd), true, false);

		ArrayList<String> mergedCnl = feClient.ToCNLList(false);

		for (String cnl : CnlToAdd)
		{
			assert mergedCnl.Any(x = cnl.equals(> x));
		}
		assert 3 == mergedCnl.size();
	}
//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//ORIGINAL LINE: [Test] public void RemoveCnl()
	public final void RemoveCnl()
	{
		CogniPySvr feClient = new CogniPySvr();
		// reason the new context
		String CnlContent = "John is a man.\r\nEvery man is a human-being.\r\n Every human-being has-name equal-to 'aaa'.\r\n";
		ArrayList<String> CnlToRemove = new ArrayList<String>(Arrays.asList("John is a man."));

		feClient.LoadCnlFromString(CnlContent, true, true, false);
		feClient.KnowledgeDelete(tangible.StringHelper.join("\r\n", CnlToRemove), false);
		ArrayList<String> mergedCnl = feClient.ToCNLList(true, true, true);

		for (String cnl : CnlToRemove)
		{
			assert!mergedCnl.Any(x = cnl.equals(> x));
		}
		assert 2 == mergedCnl.size();
	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//ORIGINAL LINE: [Test] public void RulesKeptAfterMergeTest()
	public final void RulesKeptAfterMergeTest()
	{
		CogniPySvr feClient = new CogniPySvr();
		// reason the new context
		ArrayList<String> CnlContent = new ArrayList<String>(Arrays.asList("If a man is a human-being then the man is a cat."));
		ArrayList<String> CnlToAdd = new ArrayList<String>(Arrays.asList("John is a man."));

		feClient.LoadCnlFromString(tangible.StringHelper.join("\r\n", CnlContent), true, true, false);
		feClient.KnowledgeInsert(tangible.StringHelper.join("\r\n", CnlToAdd), true, true);
		ArrayList<String> mergedCnl = feClient.ToCNLList(true, true, true);

		assert 2 == mergedCnl.size();
		for (String cnl : CnlContent)
		{
			assert mergedCnl.Any(x = cnl.equals(> x));
		}

		for (String cnl : CnlToAdd)
		{
			assert mergedCnl.Any(x = cnl.equals(> x));
		}

	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//ORIGINAL LINE: [Test][TestCase("John", "instance")][TestCase("man", "concept")][TestCase("has-name", "role")][TestCase("John[TIS]", "instance")] public void FromToUriTest(string cnlName, string type)
	public final void FromToUriTest(String cnlName, String type)
	{
		CogniPySvr feClient = new CogniPySvr();
		feClient.LoadCnlFromString("John is a man.\r\nReferences: [TIS] (\"http://www.somenamespace.com/\").", true, false, false);

		String uri = feClient.UriFromCnl(cnlName, type);
		String transformedCnl = feClient.CnlFromUri(uri, type);

		assert cnlName == transformedCnl;
	}
}